### PR TITLE
Windows Node Group Support, Upgrade Grafana Operator, Add SecretRef to Flux and Upgrade VPC CNI.

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*
This PR takes care of Upgrading Grafana Operator, Adding SecretRef to Flux and Upgrading VPC CNI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shapirov103 — COMMENTED** [review]
@elamaran11 do you mind outlining a solution with blueprints that shows how to create that secret in the first place, make it a dependency for the flux add-on and inject?

**Comment by elamaran11** [comment]
> @elamaran11 do you mind outlining a solution with blueprints that shows how to create that secret in the first place, make it a dependency for the flux add-on and inject?

@shapirov103 Added doc update for this comment.

File: docs/addons/fluxcd.md [hunk 1]
```diff
@@ -29,6 +29,10 @@ const blueprint = blueprints.EksBlueprint.builder()
   .build(app, 'my-stack-name');
 ```
 
+## Secret Management for private Git repositories
```
Comment (shapirov103) [inline docs/addons/fluxcd.md:32]: Have you tried it? I am not sure how exactly dependency is handled between aws provider, secret creation and flux installation.

**Review by shapirov103 — COMMENTED** [review]
One comment

File: examples/blueprint-construct/index.ts [hunk 2]
```diff
@@ -199,51 +196,9 @@ export default class BlueprintConstruct {
...
-                            'us-west-2': 'ami-0403ff342ceb30967',
-                            'us-east-2': 'ami-07109d69738d6e1ee',
-                            'us-west-1': 'ami-07bda4b61dc470985',
-                            'us-gov-west-1': 'ami-0e9ebbf0d3f263e9b',
-                            'us-gov-east-1':'ami-033eb9bc6daf8bfb1'
-                        }),
-                        userData: userData,
-                    }
-                }
+                addGenericNodeGroup(),
```
Comment (shapirov103) [inline examples/blueprint-construct/index.ts:199]: Nice, thank you for doing this!

**Review by shapirov103 — COMMENTED** [review]
Make sure `make lint` is success.

File: examples/blueprint-construct/index.ts (hunk 2 above)
Comment (elamaran11) [reply examples/blueprint-construct/index.ts:199]: Thanks.

File: docs/addons/fluxcd.md (hunk 1 above)
Comment (elamaran11) [reply docs/addons/fluxcd.md:32]: Yeah, i did try to create secret with external secrets to access a repo.

**Comment by elamaran11** [comment]
> Make sure `make lint` is success.

Fixed all lint issues,

File: docs/addons/fluxcd.md [hunk 3]
```diff
@@ -13,22 +13,168 @@ import * as blueprints from '@aws-quickstart/eks-blueprints';
...
+kind: Secret
+metadata:
+  name: repository-creds
+  namespace: flux-system
+type: Opaque
+data:
+  bearerToken: <<BASE64 ENCODED SECRET OF YOUR_GIT_BEARER_TOKEN STORE in AWS SECRETS MANAGER>>
+```
+
+As pointed, If you are looking to use `secretRef` to reference a secret for FluxCD addon to sync private Git repos, please make sure the referenced secret is already created in the namespace ahead of time in AWS Secrets Manager as shown above. You can use [External Secrets Addon](./external-secrets.md) to learn more about external secrets operator which allows integration with third-party secret stores like AWS Secrets Manager, AWS Systems Manager Parameter Store and inject the values into the EKS cluster as Kubernetes Secrets.
```
Comment (shapirov103) [inline docs/addons/fluxcd.md:176]: minor: If => if

File: lib/addons/vpc-cni/index.ts [hunk 4]
```diff
@@ -198,7 +292,7 @@ export interface CustomNetworkingConfig {
 
 const defaultProps: CoreAddOnProps = {
   addOnName: 'vpc-cni',
-  version: 'v1.12.6-eksbuild.2',
+  version: 'v1.13.0-eksbuild.1',
```
Comment (shapirov103) [inline lib/addons/vpc-cni/index.ts:295]: I am a bit concerned why we are using this particular version. 

The latest is 1.13.2. The version that you used is not default for any of the exiting supported versions. v1.12.6-eksbuild.2 is default for 1.27. 
If you intend to use the latest than it should be 1.13.2. However, my recommendation is to use a version which is default for the current highest supported version.

**Review by shapirov103 — COMMENTED** [review]
Awesome work.  One minor comment and a question.

File: docs/addons/fluxcd.md (hunk 3 above)
Comment (elamaran11) [reply docs/addons/fluxcd.md]: Fixed.

File: lib/addons/vpc-cni/index.ts (hunk 4 above)
Comment (elamaran11) [reply lib/addons/vpc-cni/index.ts:295]: @shapirov103 Check this doc - https://docs.aws.amazon.com/eks/latest/userguide/managing-vpc-cni.html `v1.13.2-eksbuild.1` only version that can be used across quite few versions per the doc. SO lets stay here and update later as needed.

Comment (shapirov103) [reply lib/addons/vpc-cni/index.ts:295]: Makes sense.

**Pull request merged**